library/components/tableComponents/TSelectColumns.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <button
+    class="
+      rounded
+      cursor-pointer
+      border-1 border-[#C3C2CB]
+      text-[#555463]
+      bg-white
+      w-max
+      p-1
+    "
+    style="height: fit-content"
+    :disabled="is_loading"
+    @click="download"
+  >
+    <div class="flex items-center gap-1">
+      <t-loading-spinner v-if="is_loading" position="relative" />
+      <download-outline-icon v-else :size="18" />
+      {{ label }}
+    </div>
+  </button>
+</template>
+
+<script>
+export default {
+  name: "TSelectColumns",
+  props: {
+    tLayer: {
+      type: String,
+      required: true,
+    },
+    label: {
+      type: String,
+      default: "Download CSV",
+    },
+    additionalFilters: {
+      type: Object,
+      default: () => {},
+    },
+  },
+  data: () => ({
+    is_loading: false,
+  }),
+  methods: {
+    download() {
+      this.is_loading = true;
+      payload = {
+        layer: this.tLayer,
+      };
+      filters = this.getFiltersState ? this.getFiltersState : {};
```

Comment:
Is there any case were `getFiltersState` has returned something different than an object? It should not, but I'm interested into know if you have seen that happened